systemd: v257.9 -> v258

ElvishJerricco opened the pull request:
Closes #441918

<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform:
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- Tested, as applicable:
  - [x] [NixOS tests] in [nixos/tests].
  - [ ] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [ ] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [x] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [ ] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

ElvishJerricco committed:
nixos/users-groups: New clock system group

ElvishJerricco committed:
systemd: v257.9 -> v258

philiptaron commented on pkgs/os-specific/linux/systemd/default.nix:
```diff
@@ -197,7 +197,7 @@ assert withBootloader -> withEfi;
   wantCurl = withRemote || withImportd;
 
-  version = "257.7";
+  version = "258-rc1";
```
[v258-rc2](https://github.com/systemd/systemd/releases/tag/v258-rc2) is out!

Frontear committed:
nixos/systemd-oomd: drop old `After=` dependency

flokli commented:
258 has been released!

andrevmatos commented:
Can this go to staging before #440130, so it rebuilds only once?

ElvishJerricco committed:
nixos/tests/machinectl: Slightly changed ID shift message

ElvishJerricco committed:
nixos/tests/systemd: Slightly changed IP accounting message

arianvp commented:
Regression in journald  has been reported here: https://github.com/systemd/systemd/issues/39057

schuelermine commented:
note that this is apparently fixing an IMO serious resolved problem https://github.com/systemd/systemd/issues/38951

ElvishJerricco committed:
systemd: Patch journald's Audit setting

https://github.com/systemd/systemd/pull/39069

philiptaron commented:
I've been trying to build my system with this PR. Notes so far:

### `colord` has a build break in `udevCheckPhase`

```
/nix/store/jngkccy3ib1dg8pc9308aci7cz9zazw5-colord-1.4.6/lib/udev/rules.d/69-cd-sensors.rules:105 The line has no effect, ignoring.
/nix/store/jngkccy3ib1dg8pc9308aci7cz9zazw5-colord-1.4.6/lib/udev/rules.d/69-cd-sensors.rules: udev rules check failed.
```

ElvishJerricco committed:
udevCheckHook: Use `--resolve-names=late`

With systemd 258, `--resolve-names=never` causes any udev rules such
as `ENV{FOO}=="*?", GROUP=foo` to fail due to name resolution. Restore
old behvaior using the new `late` option.

ElvishJerricco committed:
nixos/systemd-stage-1: Add `dmi_memory_id`

It's unclear why we didn't get the error message about this binary not
existing before, because the udev rule requiring it has always been in
systemd initrd.

ElvishJerricco committed:
nixos/systemd-oomd: Drop swap ordering

This ordering is present in the upstream unit in 258

arianvp commented:
These are all the failures I I ran into:

```
error: builder for '/nix/store/fix8j3zg769hm68xbh65687d9l8m04fy-vm-test-run-systemd-journal-gateway.drv' failed with exit code 1;
       last 25 log lines:
       > client # [   17.000222] systemd-journal-gatewayd[1008]: Option --trust= is not available.
       > client # [   17.007211] systemd[1]: systemd-journal-gatewayd.service: Main process exited, code=exited, status=1/FAILURE
       > client # [   17.010541] systemd[1]: systemd-journal-gatewayd.service: Failed with result 'exit-code'.
       > client # [   17.013318] systemd[1]: Started Journal Gateway Service.
       > client # [   17.093022] systemd-journal-gatewayd[1011]: Option --trust= is not available.
       > client # [   17.100927] systemd[1]: systemd-journal-gatewayd.service: Main process exited, code=exited, status=1/FAILURE
       > client # [   17.101777] systemd[1]: systemd-journal-gatewayd.service: Failed with result 'exit-code'.
       > client # [   17.103924] systemd[1]: systemd-journal-gatewayd.service: Start request repeated too quickly.
       > client # [   17.106165] systemd[1]: systemd-journal-gatewayd.service: Failed with result 'exit-code'.
       > client # [   17.106969] systemd[1]: Failed to start Journal Gateway Service.
       > client # [   17.107623] systemd[1]: systemd-journal-gatewayd.socket: Failed with result 'service-start-limit-hit'.
       > client # [   17.108961] systemd[1]: systemd-journal-gatewayd.socket: Consumed 406us CPU time, 256K memory peak, 1.7K incoming IP traffic, 224B outgoing IP traffic.
       > client #   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
       > client # curl: (35) Recv failure: Connection reset by peer
       > client: output:
       > !!! Traceback (most recent call last):
       > !!!   File "<string>", line 48, in <module>
       > !!!     machine_info = client.succeed(f"{curl_cli} {base_url}/machine")
       > !!! 
       > !!! RequestedAssertionFailed: command `/nix/store/0gmdkrx09gwjzy9skah4v6makk136ypy-curl-8.16.0-bin/bin/curl --header "Accept: application/json" --cacert /run/secrets/ca.cert.pem --cert /run/secrets/server/cert.pem --key /run/secrets/server/key.pem --fail https://client:19531/machine` failed (exit code 35)
       > cleanup
       > kill machine (pid 23)
       > qemu-system-aarch64: terminating on signal 15 from pid 8 (/nix/store/jj007xqxislbfphgwsiirsyxyif0dgss-python3-3.13.7/bin/python3.13)
       > kill vlan (pid 9)
       > (finished: cleanup, in 0.01 seconds)
       For full logs, run 'nix log /nix/store/fix8j3zg769hm68xbh65687d9l8m04fy-vm-test-run-systemd-journal-gateway.drv'.
error: builder for '/nix/store/6h9cryhybvqzsipc6f645cfhk292jw7f-vm-test-run-systemd-boot-xbootldr.drv' failed with exit code 1;
       last 25 log lines:
       > machine # [    7.088232] dhcpcd[722]: eth0: probing address 10.0.2.15/24
       > machine # [    9.342931] dhcpcd[722]: eth0: soliciting an IPv6 router
       > machine # [    9.344181] dhcpcd[722]: eth0: Router Advertisement from fe80::2
       > machine # [    9.345459] dhcpcd[722]: eth0: adding address fec0::5054:ff:fe12:3456/64
       > machine # [    9.346810] dhcpcd[722]: eth0: adding route to fec0::/64
       > machine # [    9.347940] dhcpcd[722]: eth0: adding default route via fe80::2
       > machine # [   12.230545] dhcpcd[722]: eth0: leased 10.0.2.15 for 86400 seconds
       > machine # [   12.232670] dhcpcd[722]: eth0: adding route to 10.0.2.0/24
       > machine # [   12.234466] dhcpcd[722]: eth0: adding default route via 10.0.2.2
       > machine # [   12.413003] systemd[1]: Started DHCP Client.
       > machine # [   12.415152] systemd[1]: Reached target Multi-User System.
       > machine # [   12.415884] systemd[1]: Startup finished in 5.783s (firmware) + 5.258s (loader) + 2.028s (kernel) + 10.386s (userspace) = 23.457s.
       > machine: (finished: waiting for unit multi-user.target, in 24.08 seconds)
       > machine: must succeed: test -e /efi/EFI/systemd/systemd-bootx64.efi
       > machine: output:
       > !!! Traceback (most recent call last):
       > !!!   File "<string>", line 26, in <module>
       > !!!     machine.succeed("test -e /efi/EFI/systemd/systemd-bootx64.efi")
       > !!! 
       > !!! RequestedAssertionFailed: command `test -e /efi/EFI/systemd/systemd-bootx64.efi` failed (exit code 1)
       > cleanup
       > kill machine (pid 14)
       > qemu-system-aarch64: terminating on signal 15 from pid 8 (/nix/store/jj007xqxislbfphgwsiirsyxyif0dgss-python3-3.13.7/bin/python3.13)
       > kill vlan (pid 9)
       > (finished: cleanup, in 0.01 seconds)
       For full logs, run 'nix log /nix/store/6h9cryhybvqzsipc6f645cfhk292jw7f-vm-test-run-systemd-boot-xbootldr.drv'.
error: builder for '/nix/store/f1aa0d0nrynlk8rlbjhpn86fv8bqhgs1-vm-test-run-systemd-boot-update.drv' failed with exit code 1;
       last 25 log lines:
       > machine # [    3.832238] systemd[1]: Finished Permit User Sessions.
       > machine # [    3.836221] systemd[1]: Started Getty on tty1.
       > machine # [    3.836661] systemd[1]: Reached target Login Prompts.
       > machine # [    4.272485] input: QEMU Virtio Keyboard as /devices/pci0000:00/0000:00:07.0/virtio6/input/input3
       > machine: (finished: must succeed: /run/current-system/bin/switch-to-configuration boot 2>&1, in 0.88 seconds)
       > cleanup
       > kill machine (pid 11)
       > qemu-system-aarch64: terminating on signal 15 from pid 8 (/nix/store/jj007xqxislbfphgwsiirsyxyif0dgss-python3-3.13.7/bin/python3.13)
       > kill vlan (pid 9)
       > (finished: cleanup, in 0.01 seconds)
       > Traceback (most recent call last):
       >   File "/nix/store/myrxl4hfkv3xs4dfzk1w6g4h0zjd8j0h-nixos-test-driver-1.1/bin/.nixos-test-driver-wrapped", line 9, in <module>
       >     sys.exit(main())
       >              ~~~~^^
       >   File "/nix/store/myrxl4hfkv3xs4dfzk1w6g4h0zjd8j0h-nixos-test-driver-1.1/lib/python3.13/site-packages/test_driver/__init__.py", line 162, in main
       >     driver.run_tests()
       >     ~~~~~~~~~~~~~~~~^^
       >   File "/nix/store/myrxl4hfkv3xs4dfzk1w6g4h0zjd8j0h-nixos-test-driver-1.1/lib/python3.13/site-packages/test_driver/driver.py", line 246, in run_tests
       >     self.test_script()
       >     ~~~~~~~~~~~~~~~~^^
       >   File "/nix/store/myrxl4hfkv3xs4dfzk1w6g4h0zjd8j0h-nixos-test-driver-1.1/lib/python3.13/site-packages/test_driver/driver.py", line 204, in test_script
       >     exec(self.tests, symbols, None)
       >     ~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^
       >   File "<string>", line 15, in <module>
       > AssertionError: systemd-boot not copied to to /boot/EFI/systemd/systemd-bootx64.efi
       For full logs, run 'nix log /nix/store/f1aa0d0nrynlk8rlbjhpn86fv8bqhgs1-vm-test-run-systemd-boot-update.drv'.
error: builder for '/nix/store/m1hb5lh5g5ifipw1gc943inh22bldmna-vm-test-run-systemd-lock-handler.drv' failed with exit code 143;
       last 18 log lines:
       > Machine state will be reset. To keep it, pass --keep-vm-state
       > start all VLans
       > start vlan
       > running vlan (pid 9; ctl /build/vde1.ctl)
       > (finished: start all VLans, in 0.00 seconds)
       > Test will time out and terminate in 3600 seconds
       > run the VM test script
       > additionally exposed symbols:
       >     machine,
       >     vlan1,
       >     start_all, test_script, machines, vlans, driver, log, os, create_machine, subtest, run_tests, join_all, retry, serial_stdout_off, serial_stdout_on, polling_condition, Machine, t, debug
       > machine: waiting for unit graphical.target
       > machine: waiting for the VM to finish booting
       > machine: starting vm
       > mke2fs 1.47.3 (8-Jul-2025)
       > qemu-system-aarch64: Virtio VGA not available
       > timeout reached; test terminating...
       > Terminated                 /nix/store/x4llhw6s9zi2fnaca2kckhk5xrq8a9rz-nixos-test-driver-systemd-lock-handler/bin/nixos-test-driver -o $out
       For full logs, run 'nix log /nix/store/m1hb5lh5g5ifipw1gc943inh22bldmna-vm-test-run-systemd-lock-handler.drv'.
error: builder for '/nix/store/kf9dvsh5sl38jrhyajf90v7h4vrfvmk9-vm-test-run-systemd-boot-specialisation.drv' failed with exit code 1;
       last 25 log lines:
       > machine # [   10.524522] dhcpcd[696]: eth0: leased 10.0.2.15 for 86400 seconds
       > machine # [   10.526522] dhcpcd[696]: eth0: adding route to 10.0.2.0/24
       > machine # [   10.527698] dhcpcd[696]: eth0: adding default route via 10.0.2.2
       > machine # [   10.606945] systemd[1]: Started DHCP Client.
       > machine # [   10.608892] systemd[1]: Reached target Multi-User System.
       > machine # [   10.610193] systemd[1]: Startup finished in 5.754s (firmware) + 5.210s (loader) + 1.244s (kernel) + 9.365s (userspace) = 21.573s.
       > machine: (finished: waiting for unit multi-user.target, in 22.11 seconds)
       > machine: must succeed: test -e /boot/loader/entries/nixos-generation-1-specialisation-something.conf
       > machine: (finished: must succeed: test -e /boot/loader/entries/nixos-generation-1-specialisation-something.conf, in 0.01 seconds)
       > machine: must succeed: grep -q 'title NixOS (something)' /boot/loader/entries/nixos-generation-1-specialisation-something.conf
       > machine: (finished: must succeed: grep -q 'title NixOS (something)' /boot/loader/entries/nixos-generation-1-specialisation-something.conf, in 0.02 seconds)
       > machine: must succeed: grep 'sort-key something' /boot/loader/entries/nixos-generation-1-specialisation-something.conf
       > machine: (finished: must succeed: grep 'sort-key something' /boot/loader/entries/nixos-generation-1-specialisation-something.conf, in 0.02 seconds)
       > machine: must succeed: grep 'devicetree /EFI/nixos/[a-z0-9]\{32\}.*dummy' /boot/loader/entries/nixos-generation-1-specialisation-something.conf
       > machine: output:
       > !!! Traceback (most recent call last):
       > !!!   File "<string>", line 13, in <module>
       > !!!     machine.succeed(
       > !!! 
       > !!! RequestedAssertionFailed: command `grep 'devicetree /EFI/nixos/[a-z0-9]\{32\}.*dummy' /boot/loader/entries/nixos-generation-1-specialisation-something.conf` failed (exit code 1)
       > cleanup
       > kill machine (pid 11)
       > qemu-system-aarch64: terminating on signal 15 from pid 8 (/nix/store/jj007xqxislbfphgwsiirsyxyif0dgss-python3-3.13.7/bin/python3.13)
       > kill vlan (pid 9)
       > (finished: cleanup, in 0.01 seconds)
       For full logs, run 'nix log /nix/store/kf9dvsh5sl38jrhyajf90v7h4vrfvmk9-vm-test-run-systemd-boot-specialisation.drv'.
error: builder for '/nix/store/lsx8dh222y0v1li1xfwbrxj5ywv8d7if-vm-test-run-systemd-initrd-luks-unl0kr.drv' failed with exit code 143;
       last 17 log lines:
       > Machine state will be reset. To keep it, pass --keep-vm-state
       > start all VLans
       > start vlan
       > running vlan (pid 9; ctl /build/vde1.ctl)
       > (finished: start all VLans, in 0.00 seconds)
       > Test will time out and terminate in 3600 seconds
       > run the VM test script
       > additionally exposed symbols:
       >     machine,
       >     vlan1,
       >     start_all, test_script, machines, vlans, driver, log, os, create_machine, subtest, run_tests, join_all, retry, serial_stdout_off, serial_stdout_on, polling_condition, Machine, t, debug
       > machine: waiting for unit multi-user.target
       > machine: waiting for the VM to finish booting
       > machine: starting vm
       > qemu-system-aarch64: Virtio VGA not available
       > timeout reached; test terminating...
       > Terminated                 /nix/store/li0k9ghsik4vlbl6clva2mllp9w5zdfa-nixos-test-driver-systemd-initrd-luks-unl0kr/bin/nixos-test-driver -o $out
       For full logs, run 'nix log /nix/store/lsx8dh222y0v1li1xfwbrxj5ywv8d7if-vm-test-run-systemd-initrd-luks-unl0kr.drv'.
error: builder for '/nix/store/apqk9ky6iq8q60wr4yyw8bck4a9vf27v-vm-test-run-systemd-initrd-luks-tpm2.drv' failed with exit code 143;
       last 25 log lines:
       > machine # [    1.335042] systemd[1]: Starting Dispatch Password Requests to Console...
       > machine # [    1.394876] systemd[1]: Started Dispatch Password Requests to Console.
       > machine # [    1.404449] systemd-tty-ask-password-agent[126]: Starting password query on /dev/tty1.
       > machine # Please enter passphrase for disk cryptroot: (press TAB for no echo) [    1.409265] systemd-tty-ask-password-agent[125]: Starting password query on /dev/ttyAMA0.
       > machine # [    1.790255] tpm_tis MSFT0101:00: 2.0 TPM (device-id 0x1, rev-id 1)
       > machine # [    1.794154] tpm tpm0: A TPM error (256) occurred attempting the self test
       > machine # [    1.794656] tpm tpm0: starting up the TPM manually
       > machine # [    1.862818] systemd[1]: systemd-vconsole-setup.service: Deactivated successfully.
       > machine # [    1.863433] systemd[1]: Stopped Virtual Console Setup.
       > machine # [    1.863798] systemd[1]: Stopping Virtual Console Setup...
       > machine # [    1.871985] systemd[1]: Starting Virtual Console Setup...
       > machine # [    1.874840] systemd[1]: systemd-vconsole-setup.service: Deactivated successfully.
       > machine # [    1.876152] systemd[1]: Stopped Virtual Console Setup.
       > machine # [    1.881353] systemd[1]: Starting Virtual Console Setup...
       > machine # [    1.909162] systemd[1]: Finished Virtual Console Setup.
       > machine # [    1.957581] systemd[1]: Condition check resulted in /dev/tpmrm0 being skipped.
       > machine # [    1.959329] systemd[1]: Expecting device /dev/tpm0...
       > machine # [    1.959774] systemd[1]: Found device /dev/tpm0.
       > machine # [    1.960735] systemd[1]: Reached target Trusted Platform Module.
       > timeout reached; test terminating...
       > kill machine (pid 64)
       > qemu-system-aarch64: terminating on signal 15 from pid 8 (/nix/store/jj007xqxislbfphgwsiirsyxyif0dgss-python3-3.13.7/bin/python3.13)
       > cleanup
       > kill machine (pid 64)
       > Terminated                 /nix/store/gal9ld5qy0jjycx4a6klgaww3qmr4qxl-nixos-test-driver-systemd-initrd-luks-tpm2/bin/nixos-test-driver -o $out
       For full logs, run 'nix log /nix/store/apqk9ky6iq8q60wr4yyw8bck4a9vf27v-vm-test-run-systemd-initrd-luks-tpm2.drv'.
error: builder for '/nix/store/mggjw1r60lrl7s0ffx0iwmjzfld0djs8-vm-test-run-systemd-repart-after-initrd.drv' failed with exit code 1;
       last 25 log lines:
       > machine # [   13.224842] systemd[1]: Reached target Multi-User System.
       > machine # [   13.225453] systemd[1]: Startup finished in 2.273s (kernel) + 10.951s (userspace) = 13.224s.
       > machine: (finished: waiting for unit multi-user.target, in 14.31 seconds)
       > machine: must succeed: journalctl --unit systemd-repart.service
       > machine: (finished: must succeed: journalctl --unit systemd-repart.service, in 0.06 seconds)
       > cleanup
       > kill machine (pid 14)
       > qemu-system-aarch64: terminating on signal 15 from pid 8 (/nix/store/jj007xqxislbfphgwsiirsyxyif0dgss-python3-3.13.7/bin/python3.13)
       > kill vlan (pid 9)
       > (finished: cleanup, in 0.01 seconds)
       > Traceback (most recent call last):
       >   File "/nix/store/myrxl4hfkv3xs4dfzk1w6g4h0zjd8j0h-nixos-test-driver-1.1/bin/.nixos-test-driver-wrapped", line 9, in <module>
       >     sys.exit(main())
       >              ~~~~^^
       >   File "/nix/store/myrxl4hfkv3xs4dfzk1w6g4h0zjd8j0h-nixos-test-driver-1.1/lib/python3.13/site-packages/test_driver/__init__.py", line 162, in main
       >     driver.run_tests()
       >     ~~~~~~~~~~~~~~~~^^
       >   File "/nix/store/myrxl4hfkv3xs4dfzk1w6g4h0zjd8j0h-nixos-test-driver-1.1/lib/python3.13/site-packages/test_driver/driver.py", line 246, in run_tests
       >     self.test_script()
       >     ~~~~~~~~~~~~~~~~^^
       >   File "/nix/store/myrxl4hfkv3xs4dfzk1w6g4h0zjd8j0h-nixos-test-driver-1.1/lib/python3.13/site-packages/test_driver/driver.py", line 204, in test_script
       >     exec(self.tests, symbols, None)
       >     ~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^
       >   File "<string>", line 27, in <module>
       > AssertionError
       For full logs, run 'nix log /nix/store/mggjw1r60lrl7s0ffx0iwmjzfld0djs8-vm-test-run-systemd-repart-after-initrd.drv'.
error: builder for '/nix/store/lj28nlw870vqnqfaafma9rb4p1m1h5xx-vm-test-run-installer-simpleUefiGrub.drv' failed with exit code 1;
       last 25 log lines:
       > subtest: Do it again to make sure it's idempotent
       > installer: must succeed: nixos-install < /dev/null >&2
       > installer # copying channel...
       > installer # building the configuration in /mnt/etc/nixos/configuration.nix...
       > installer # /nix/store/9wa8ilgaxynhs1jhrrd2xycvy8g3ddsj-nixos-system-thatworked-25.11test
       > installer # installing the boot loader...
       > installer # setting up /etc...
       > installer # updating GRUB 2 menu...
       > installer # installing the GRUB 2 boot loader into /boot...
       > installer # Installing for arm64-efi platform.
       > installer # /nix/store/6ps79z99ba3ccrvxx3zwrlj9vmdr5gws-grub-2.12/sbin/grub-install: error: cannot copy `/nix/store/6ps79z99ba3ccrvxx3zwrlj9vmdr5gws-grub-2.12/lib/grub/arm64-efi/test.mod' to `/boot/grub/arm64-efi/test.mod': No space left on device.
       > installer # /nix/store/iqzy2pa1m0i286sy4kxf5vymm5a5hpam-install-grub.pl: installation of GRUB EFI into /boot failed: Inappropriate ioctl for device
       > installer # Failed to install bootloader
       > installer: output:
       > !!! Test "Do it again to make sure it's idempotent" failed with error: "command `nixos-install < /dev/null >&2` failed (exit code 1)"
       > !!! Traceback (most recent call last):
       > !!!   File "<string>", line 47, in <module>
       > !!!     installer.succeed("nixos-install < /dev/null >&2")
       > !!! 
       > !!! RequestedAssertionFailed: command `nixos-install < /dev/null >&2` failed (exit code 1)
       > cleanup
       > kill machine (pid 11)
       > qemu-system-aarch64: terminating on signal 15 from pid 8 (/nix/store/jj007xqxislbfphgwsiirsyxyif0dgss-python3-3.13.7/bin/python3.13)
       > kill vlan (pid 9)
       > (finished: cleanup, in 0.01 seconds)
       For full logs, run 'nix log /nix/store/lj28nlw870vqnqfaafma9rb4p1m1h5xx-vm-test-run-installer-simpleUefiGrub.drv'.
error: builder for '/nix/store/wznga1sn7qx2fjlqzmg5xw5q1hqc4xq5-vm-test-run-installer-simpleUefiGrubSpecialisation.drv' failed with exit code 1;
       last 25 log lines:
       > subtest: Do it again to make sure it's idempotent
       > installer: must succeed: nixos-install < /dev/null >&2
       > installer # copying channel...
       > installer # building the configuration in /mnt/etc/nixos/configuration.nix...
       > installer # /nix/store/rahxcxadmxdawl71hyq5nqb6xjbhxbps-nixos-system-thatworked-25.11test
       > installer # installing the boot loader...
       > installer # setting up /etc...
       > installer # updating GRUB 2 menu...
       > installer # installing the GRUB 2 boot loader into /boot...
       > installer # Installing for arm64-efi platform.
       > installer # /nix/store/6ps79z99ba3ccrvxx3zwrlj9vmdr5gws-grub-2.12/sbin/grub-install: error: cannot copy `/nix/store/6ps79z99ba3ccrvxx3zwrlj9vmdr5gws-grub-2.12/lib/grub/arm64-efi/minix3.mod' to `/boot/grub/arm64-efi/minix3.mod': No space left on device.
       > installer # /nix/store/iqzy2pa1m0i286sy4kxf5vymm5a5hpam-install-grub.pl: installation of GRUB EFI into /boot failed: Inappropriate ioctl for device
       > installer # Failed to install bootloader
       > installer: output:
       > !!! Test "Do it again to make sure it's idempotent" failed with error: "command `nixos-install < /dev/null >&2` failed (exit code 1)"
       > !!! Traceback (most recent call last):
       > !!!   File "<string>", line 47, in <module>
       > !!!     installer.succeed("nixos-install < /dev/null >&2")
       > !!! 
       > !!! RequestedAssertionFailed: command `nixos-install < /dev/null >&2` failed (exit code 1)
       > cleanup
       > kill machine (pid 11)
       > qemu-system-aarch64: terminating on signal 15 from pid 8 (/nix/store/jj007xqxislbfphgwsiirsyxyif0dgss-python3-3.13.7/bin/python3.13)
       > kill vlan (pid 9)
       > (finished: cleanup, in 0.01 seconds)
       For full logs, run 'nix log /nix/store/wznga1sn7qx2fjlqzmg5xw5q1hqc4xq5-vm-test-run-installer-simpleUefiGrubSpecialisation.drv'.
error: builder for '/nix/store/nx4gi0bx7p7gp8g0js18hij2zy7db2j7-vm-test-run-systemd-initrd-vconsole.drv' failed with exit code 143;
       last 25 log lines:
       >     vlan1,
       >     start_all, test_script, machines, vlans, driver, log, os, create_machine, subtest, run_tests, join_all, retry, serial_stdout_off, serial_stdout_on, polling_condition, Machine, t, debug
       > machine: starting vm
       > mke2fs 1.47.3 (8-Jul-2025)
       > machine: QEMU running (pid 11)
       > machine # Disk image does not exist, creating the virtualisation disk image...
       > machine: waiting for Press Enter for maintenance to appear on console
       > machine # Formatting '/build/vm-state-machine/tmp.iWqUggMjcc', fmt=raw size=1073741824
       > machine # Discarding device blocks: done         
       > machine # Creating filesystem with 262144 4k blocks and 65536 inodes
       > machine # Filesystem UUID: df6a87c3-8568-468a-a9be-b87d0e67987b
       > machine # Superblock backups stored on blocks:
       > machine #        32768, 98304, 163840, 229376
       > machine # 
       > machine # Allocating group tables: done
       > machine # Writing inode tables: done
       > machine # Creating journal (8192 blocks): done
       > machine # Writing superblocks and filesystem accounting information: done
       > machine # 
       > machine # Virtualisation disk image created.
       > machine # [    1.814800] (udev-worker)[100]: mtd0ro: Failed to find and pin callout binary "/nix/store/g60fbiim1jnsqi7sayvr3ld8pkii2fpg-systemd-258/lib/udev/mtd_probe": No such file or directory
       > timeout reached; test terminating...
       > kill machine (pid 11)
       > qemu-system-aarch64: terminating on signal 15 from pid 8 (/nix/store/jj007xqxislbfphgwsiirsyxyif0dgss-python3-3.13.7/bin/python3.13)
       > Terminated                 /nix/store/rvcw8dr386nvzshphqxn2r9d4zz1h3hc-nixos-test-driver-systemd-initrd-vconsole/bin/nixos-test-driver -o $out
       For full logs, run 'nix log /nix/store/nx4gi0bx7p7gp8g0js18hij2zy7db2j7-vm-test-run-systemd-initrd-vconsole.drv'.
error: build of '/nix/store/6h9cryhybvqzsipc6f645cfhk292jw7f-vm-test-run-systemd-boot-xbootldr.drv', '/nix/store/apqk9ky6iq8q60wr4yyw8bck4a9vf27v-vm-test-run-systemd-initrd-luks-tpm2.drv', '/nix/store/f1aa0d0nrynlk8rlbjhpn86fv8bqhgs1-vm-test-run-systemd-boot-update.drv', '/nix/store/fix8j3zg769hm68xbh65687d9l8m04fy-vm-test-run-systemd-journal-gateway.drv', '/nix/store/kf9dvsh5sl38jrhyajf90v7h4vrfvmk9-vm-test-run-systemd-boot-specialisation.drv', '/nix/store/lj28nlw870vqnqfaafma9rb4p1m1h5xx-vm-test-run-installer-simpleUefiGrub.drv', '/nix/store/lsx8dh222y0v1li1xfwbrxj5ywv8d7if-vm-test-run-systemd-initrd-luks-unl0kr.drv', '/nix/store/m1hb5lh5g5ifipw1gc943inh22bldmna-vm-test-run-systemd-lock-handler.drv', '/nix/store/mggjw1r60lrl7s0ffx0iwmjzfld0djs8-vm-test-run-systemd-repart-after-initrd.drv', '/nix/store/nx4gi0bx7p7gp8g0js18hij2zy7db2j7-vm-test-run-systemd-initrd-vconsole.drv', '/nix/store/wznga1sn7qx2fjlqzmg5xw5q1hqc4xq5-vm-test-run-installer-simpleUefiGrubSpecialisation.drv' failed

arianvp committed:
nixos/tests/systemd-initrd-luks-tpm2: Fix on aarch64-linux

Systemd wants you to run tpm unlock on a machine with the TCG
bits of OVMF enabled. If not it just bails out with

No TPM2 hardware discovered and EFI firmware does not see it either, falling back to traditional unlocking

arianvp committed:
systemd: remove systemd-lock-handler from passthru.tests

it's not an official systemd thing and we're not the maintainers of it.
Also the test is broken. But that is not our problem

arianvp commented:
Alright I tested this on `aarch64-linux` and don't see any unexpected failing tests. This seems ready to go from my side
